### PR TITLE
tkn-results: Add CLI for listing results and records.

### PR DESCRIPTION
**Pull request by wlynch**
This adds a new tkn plugin for interacting with the Results API.
For now this only implements the list methods for results and records.

I wasn't sure what the best way to go about testing the whole CLI, since
most of the complexity is setting up the client. We might want to
consider adding this as a part of the e2e tests in a future PR.

Implements #81

### CODE REVIEW

**Comment by ImJasonH** [comment]
Is the intention for this to be a CLI for *testing/playing around with* the API? If so it probably doesn't need super thorough tests, and should just be documented as test-only.

If it's not, we probably need to have a longer discussion with the tkn folks about what the relationship is between the two.

File: tools/tkn-results/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# tkn-results
+
+tkn CLI plugin for interacting with the Tekton Results API.
+
+## Installation
+
+```
+$ GOBIN=${TKN_PLUGINS_DIR:-"${HOME}/.config/tkn/plugins"} go install github.com/tektoncd/experimental/tkn-resolve
```
Comment (afrittoli) [inline tools/tkn-results/README.md:8]: Is `github.com/tektoncd/experimental/tkn-resolve` the right path?

Comment (wlynch) [reply tools/tkn-results/README.md]: Good catch! 😅  Fixed.

**Comment by wlynch** [comment]
> Is the intention for this to be a CLI for _testing/playing around with_ the API? If so it probably doesn't need super thorough tests, and should just be documented as test-only.
> 
> If it's not, we probably need to have a longer discussion with the tkn folks about what the relationship is between the two.

Short term this is to let users have a nicer mechanism to work with the Results API without needing to jump through grpc_cli hoops. There's a blurry line here between a test-only tool and the recommended way to see results on a CLI.

I agree that we should have a longer discussion about how results fits in to the existing tkn UX for pipelineruns/taskruns (particularly since results is an optional component), but for now our hope is to start prototyping with a plugin.

File: tools/tkn-results/cmd/records_list.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+				fmt.Printf("ListRecords: %v\n", err)
+				return err
+			}
+			return printproto(os.Stdout, resp, format)
+		},
+		Args: cobra.ExactArgs(1),
+	}
+)
+
+// Execute executes the root command.
```
Comment (XinruZhang) [inline tools/tkn-results/cmd/records_list.go:38]: Typo?

File: tools/tkn-results/cmd/results_list.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+				fmt.Printf("ListResults: %v\n", err)
+				return err
+			}
+			return printproto(os.Stdout, resp, format)
+		},
+		Args: cobra.ExactArgs(1),
+	}
+)
+
+// Execute executes the root command.
```
Comment (XinruZhang) [inline tools/tkn-results/cmd/results_list.go:38]: Same here

File: tools/tkn-results/cmd/records_list.go (hunk 2 above)
Comment (wlynch) [reply tools/tkn-results/cmd/records_list.go:38]: Removed.

File: tools/tkn-results/cmd/results_list.go (hunk 3 above)
Comment (wlynch) [reply tools/tkn-results/cmd/results_list.go]: Removed.

**Review by XinruZhang — APPROVED** [review]
/lgtm

**Pull request merged**